File: src/2540.minimum-common-value.2634/solution.ts
function getCommon (nums1: number[], nums2: number[]): number {
    let index1 = 0;
    let index2 = 0;
    while (index1 < nums1.length && index2 < nums2.length) {
        const diff = nums1[index1] - nums2[index2];
        if (diff > 0) {
            index2++;
        } else if (diff < 0) {
            index1++;
        } else {
            return nums1[index1];
        }
    }

    return -1;
}

File: src/2540.minimum-common-value.2634/solution.java
class Solution {
    public int getCommon(int[] nums1, int[] nums2) {
        int index1 = 0;
        int index2 = 0;
        while (index1<nums1.length && index2<nums2.length){
            int diff = nums2[index2]-nums1[index1];
            if(diff>0){
                index1++;
            }else if(diff<0){
                index2++;
            }else{
                return nums1[index1];
            }
        }
        return -1;
    }
}